consul/config_templates/consul-values.yaml

```diff
@@ -0,0 +1,1353 @@
+    # into secondary datacenters so they can federate with this datacenter. The
+    # secret contains all the information secondary datacenters need to contact
+    # and authenticate with this datacenter. This should only be set to true
+    # in your primary datacenter. The secret name is
+    # <global.name>-federation (if setting global.name), otherwise
+    # <helm-release-name>-consul-federation.
+    # Requires consul-k8s 0.15.0+.
+    createFederationSecret: false
+
+  # Resource settings for lifecycle-sidecar containers.
+  # The lifecycle sidecar ensures the Consul services are always registered with
+  # their local consul clients and is used by the ingress/terminating/mesh gateways
+  # as well as with every connect-injected service.
+  lifecycleSidecarContainer:
+    resources:
+      requests:
+        memory: "25Mi"
+        cpu: "20m"
+      limits:
+        memory: "50Mi"
+        cpu: "20m"
+
+# Server, when enabled, configures a server cluster to run. This should
+# be disabled if you plan on connecting to a Consul cluster external to
+# the Kube cluster.
+server:
+  enabled: "-"
+  image: null
+  replicas: 1
+  bootstrapExpect: 1 # Should <= replicas count
+
+  # enterpriseLicense refers to a Kubernetes secret that you have created that
+  # contains your enterprise license. It is required if you are using an
+  # enterprise binary. Defining it here applies it to your cluster once a leader
+  # has been elected. If you are not using an enterprise image
+  # or if you plan to introduce the license key via another route, then set
+  # these fields to null.
+  # Note: the job to apply license runs on both Helm installs and upgrades.
+  enterpriseLicense:
+    secretName: null
+    secretKey: null
+
+  # storage and storageClass are the settings for configuring stateful
+  # storage for the server pods. storage should be set to the disk size of
+  # the attached volume. storageClass is the class of storage which defaults
+  # to null (the Kube cluster will pick the default).
+  storage: 10Gi
+  storageClass: null
+
+  # connect will enable Connect on all the servers, initializing a CA
+  # for Connect-related connections. Other customizations can be done
+  # via the extraConfig setting.
+  connect: true
+
+  # Resource settings for Server agents.
+  # NOTE: The use of a YAML string is deprecated. Instead, set directly as a
+  # YAML map.
+  resources:
+    requests:
+      memory: "100Mi"
+      cpu: "100m"
+    limits:
+      memory: "100Mi"
+      cpu: "100m"
+
+  # updatePartition is used to control a careful rolling update of Consul
+  # servers. This should be done particularly when changing the version
+  # of Consul. Please refer to the documentation for more information.
+  updatePartition: 0
+
+  # disruptionBudget enables the creation of a PodDisruptionBudget to
+  # prevent voluntary degrading of the Consul server cluster.
+  disruptionBudget:
+    enabled: true
```

Comment:
Should we add this parameter as a condition into the resource template name `PodDisruptionsBudget`?

Reply:
Sorry, I'm not quite sure I understand. The templates are part of the Helm chart and I don't think we should change them. Furthermore, it is a condition in the corresponding template (server-disruptionbudget.yaml).

Reply:
Ok, let's clear this. I mean should we add the condition like below.

```
{{- if .Values.disruptionBudget.enabled }}
kind:PodDisruptionBudget
metadata:
.....
{{- end }}
```

Reply:
I thinks this still needs clarification. I'm not quite sure whether we're aligned, maybe we misunderstand each other?
It is not quite clear to me where you want to implement this. Also, `consul.yaml` already contains `PodDisruptionBudget` for the server.

The `consul-values.yaml` is apart from a few small changes the same as `values.yaml` from the official Consul Helm chart.
It is used to generate `consul-new.yaml` (not version controlled) with the templates from the official Helm chart. 
Then, `sed` is used to replace the namespace in `consul-new.yaml`, and generates `consul.yaml`. This is not a Helm template, and not used in conjunction with `consul-values.yaml`. It is used as a template in consul.go (executeTemplate) to merge the namespace specified by the user. 
We should stick to the official Consul Helm chart and its templates, and not change them (if that is was what you suggested?). Neither should we introduce extra transformations. It should be very easy for us to upgrade when new chart versions are released. If we think that something is missing from the Helm chart, we can raise an issue with Hashicorp Consul. The extra transformation with `sed` here won't be necessary as soon as we have implemented Helm support.  

This comment also applies to the other comments.

Sorry if I have stated the obvious, it also helps me to think through it for myself :-)

Reply:
Haha, Do not say sorry for stating the obvious. I know your mean that this PR is not a Helm template and it's rendered from the official Helm chart we should not change for easy upgrade it that's good. But it looks like that some `key-value` not useful in this template that makes me bewilderment.

Reply:
Good, thanks :-) Asynchronous written communication is not always easy... Anyway, I've had clients that had to change official charts - for good reasons, but it was a maintenance burden, and they wished the didn't have to change them (sometimes, the same changes showed up in the official chart not much later). I just don't think we should. If we change configuration,  we'll have to document it, explain it, and maintain it. I'd rather raise an issue with Hashicorp :-)